lib/optimizely/optimizely_config.rb

```diff
@@ -16,56 +16,111 @@
 #
 
 module Optimizely
+  require 'json'
   class OptimizelyConfig
+    include Optimizely::ConditionTreeEvaluator
     def initialize(project_config)
       @project_config = project_config
+      @rollouts = @project_config.rollouts
+      @audiences = []
+      type_audiences = @project_config.typed_audiences
+      optly_typed_audiences = []
```

Comment:
whats the differece between `type_audiences` and `optly_typed_audiences`. It should be clear from the variable names that why there are two arrays being used. Actually you dont even need to use `type_audiences` one. You can use `#project_config.typed_audiences` directly to loop below. `optly_typed_audiences` does not sound consistent with others as no other variable are prefixed with `optly`. You should use `typed_audiences` to hold temp audiences and directly reference `@project_config.typed_audiences` to loop through the original array.